### PR TITLE
Quad regularization repack

### PR DESCRIPTION
**Pull request by leclere**
- add quadratic regularization
- refactoring of run_sddp function

### CODE REVIEW

**Comment by frapac** [comment]
Ok to merge.
I will take care of coverage later

File: src/objects.jl [hunk 1]
```diff
@@ -45,6 +45,8 @@ type LinearSPModel <: SPModel
     equalityConstraints::Union{Void, Function}
     inequalityConstraints::Union{Void, Function}
 
+    refTrajectories::Union{Void, Array{Float64, 3}}
```
Comment (leclere) [inline src/objects.jl:48]: That can be only temporary : an SPmodel cannot contain a "refTrajectories" which only represent an algorithm dependant property and not a stochastic model property.

File: src/stoppingtest.jl [hunk 2]
```diff
@@ -17,10 +28,11 @@ function test_stopping_criterion(param::SDDPparameters, stats::SDDPStat)
     lb = stats.lower_bounds[end]
     ub = stats.upper_bounds[end]
     check_gap = (abs((ub-lb)/lb) < param.gap)
-    check_iter = stats.niterations > param.maxItNumber
+    check_iter = stats.niterations >= param.maxItNumber
```
Comment (leclere) [inline src/stoppingtest.jl:31]: Actually when the test is done we have updated niterations. So I'm changing back to strict inequality.

Comment (frapac) [reply src/stoppingtest.jl:31]: Ok, then you could move the place where niterations is incremented.

Comment (frapac) [reply src/stoppingtest.jl:31]: I disagree! SDDPStats is 0-indexed, and you want to stop the algorithm in a 1-indexed manner. Please fix this issue. If user specifies n iterations to run SDDP, he does not want to compute n + 1 iterations.

**Pull request merged**